### PR TITLE
Ability to use template databases

### PR DESCRIPTION
**Pull request by fizyk**

### CODE REVIEW

File: src/pytest_postgresql/factories/process.py [hunk 1]
```diff
@@ -151,118 +129,25 @@ def postgresql_proc_fixture(
...
+            with DatabaseJanitor(
+                user=postgresql_executor.user,
+                host=postgresql_executor.host,
+                port=postgresql_executor.port,
+                db_name=template_db_name,
+                version=postgresql_executor.version,
+                password=postgresql_executor.password,
+            ):
+                if populate_template:
+                    for populate_function in populate_template:
```
Comment (fizyk) [inline src/pytest_postgresql/factories/process.py:142]: How many databases can be set up? Actually consider having a `Dict[str, Callable]` for `populate_template` and one function per dbname

**Review by efagerberg — COMMENTED** [review]
This is my first time looking at any code in this library so I am not a total expert. But from what I read this seems reasonable.

Is the idea now that time spent initializing a db is reduced by using the template, but we still initialize a new db each test?

If so in my experience I still would probably hit the db startup error every so often since that is happening even before I initialize the db state.

Either way this would be useful as a plugin for telling this library the expected db starting state.

**Comment by fizyk** [comment]
@efagerberg yes, the db is still initialised each test, but it should not take longer than rollback a multilevel transaction. 
That approach cleans everything to the initial state, leaving you to play with all levels (or not) of transactions or even trigger database migrations (alembic for sqlalchemy?)

File: src/pytest_postgresql/factories/process.py (hunk 1 above)
Comment (fizyk) [reply src/pytest_postgresql/factories/process.py:145]: Not at the moment, that would require a bit more rethinking though

**Comment by fizyk** [comment]
@jcpunk , since you were the author of the original code, would you like to take a look at the extension of the idea here?

**Review by jcpunk — APPROVED** [review]
Looks good to me

**Pull request merged**